### PR TITLE
GitHub Actions: get username if we don't have it

### PR DESCRIPTION
**Pull request by maxim-belkin**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This is a fix for my previous submission #6723 for cases when `api_credentials` has a token only without a username

@dawidd6 and @chenrui333, could you please set this PR with your workflow(s)?

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
> @dawidd6 and @chenrui333, could you please set this PR with your workflow(s)?

Presumable you mean "test" here? Will wait for one or both to confirm.

**Comment by dawidd6** [comment]
Can't really test it, cause this function is no longer in use. See changes in https://github.com/Homebrew/brew/pull/6770/files. You can bring it back here tho and I'll make sure to try it.

**Comment by maxim-belkin** [comment]
> > @dawidd6 and @chenrui333, could you please set this PR with your workflow(s)?
> 
> Presumable you mean "test" here? Will wait for one or both to confirm.

Yes, "test" not "set" :) 



> Can't really test it, cause this function is no longer in use. See changes in https://github.com/Homebrew/brew/pull/6770/files. You can bring it back here tho and I'll make sure to try it.

done

**Comment by dawidd6** [comment]
Yea, it works now: https://github.com/Homebrew/homebrew-core/pull/47229

File: Library/Homebrew/dev-cmd/bump-formula-pr.rb [hunk 1]
```diff
@@ -381,14 +381,13 @@ def bump_formula_pr
   end
 
   def forked_repo_info(formula, tap_full_name, backup_file)
-    begin
```
Comment (dawidd6) [inline Library/Homebrew/dev-cmd/bump-formula-pr.rb]: Isn't `begin` needed?

**Comment by maxim-belkin** [comment]
Nah. def ~ begin

Thanks for checking, Dawid!

**Comment by maxim-belkin** [comment]
> Yea, it works now: [Homebrew/homebrew-core#47229](https://github.com/Homebrew/homebrew-core/pull/47229)
> 
> --- nice branch name @maxim-belkin

:trollface: Ah, I got your username wrong (double `d`s _vs._ `6`s) -- sorry about that.

**Comment by dawidd6** [comment]
images

**Comment by chenrui333** [comment]
@maxim-belkin it works well for me on `brew bump-formula-pr`

Made couple of PRs just now. :) Thanks for getting this into my attention!

File: Library/Homebrew/dev-cmd/bump-formula-pr.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/bump-formula-pr.rb]: Not any more; shiny new Ruby syntax.

**Comment by maxim-belkin** [comment]
Would it be useful to factor out code that determines GitHub username?

**Comment by MikeMcQuaid** [comment]
Thanks again @maxim-belkin!

**Pull request merged**